scripts/tile_benchmark/benchmark.py

```diff
@@ -0,0 +1,365 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+#############################################################################
+#  Copyright Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#############################################################################
+
+
+import math
+import sys
+
+import matplotlib.pyplot as plt
+import numpy as np
+import requests
+
+np.set_printoptions(threshold=np.nan)
+
+URL = 0
+NUM_LEVELS = 1
+NUM_X_TILES = 2
+NUM_Y_TILES = 3
+ATLAS_ARR = 4
+
+
+def main():
+    """Read in request data from source.txt, request tiles from each server and
+    visualize response times
+    """
+    file_name = 'source.txt'
+    run_params = {}
+    # read in servers to request
+    with open(file_name, 'r') as params:
+        for line in params:
+            if '#' not in line:
+                args = line.split()
+                run_params[args[0]] = args
+    # get elapsed time for data requests
+    for host in run_params.keys():
+        if host == 'atlas':
+            print "Getting images from atlas"
+            elapsed_time = get_data(
+                "atlas",
+                'atlas hardcoded img',
+                int(run_params[host][1]),
+                int(run_params[host][2]))
+        elif host == 'adrc':
+            print "Getting images from adrc"
+            elapsed_time = get_data(
+                "adrc",
+                'adrc hardcoded img',
+                int(run_params[host][1]),
+                int(run_params[host][2]))
+        else:
+            print"Getting images from girder"
+            elapsed_time = get_data(
+                run_params[host][0],
+                run_params[host][1],
+                int(run_params[host][2]),
+                int(run_params[host][3]))
+        run_params[host].append(elapsed_time)
+
+    plot_data_and_save(run_params)
+
+
+def get_data(host, image_id, requested_levels, requested_num_tiles):
+    """get request information from each server an request tiles"""
+    num_levels = -1
+    data = {}
+    # get data about server, host url and params
+    if host == "atlas":
+
+        data = atlas_init()
+    elif host == "adrc":
+
```

Comment:
Brian are there any graphs/numbers you can share before today's meeting?

On Wed, Jun 1, 2016 at 11:44 AM Brian Helba <notifications@github.com>
wrote:

> In scripts/tile_benchmark/benchmark.py
> <https://github.com/DigitalSlideArchive/large_image/pull/59#discussion_r65387168>
> :
>
> > +                int(run_params[host][3]))
> > +        run_params[host].append(
>
> elapsed_time)
> > +
> > +    plot_data_and_save(run_params)
> > +
> > +
> > +def get_data(host, image_id, requested_levels, requested_num_tiles):
> > +    """get request information from each server an request tiles"""
> > +    num_levels = -1
> > +    data = {}
> > +    # get data about server, host url and params
> > +    if host == "atlas":
> > +
> > +        data = atlas_init()
> > +    elif host == "adrc":
> > +
>
> Or here.
>
> —
> You are receiving this because you are subscribed to this thread.
>
> Reply to this email directly, view it on GitHub
> <https://github.com/DigitalSlideArchive/large_image/pull/59/files/c774b39c5953fed6334ade91842435860ec750af#r65387168>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AArhzl4n14CWRXy7ZoMLa96zMMrRJS1Nks5qHajzgaJpZM4Inpvw>
> .
>
-- 
David A Gutman MD PhD
Assistant Professor of Neurology, Psychiatry & Biomedical Informatics
Emory University School of Medicine
Staff Physician, Mental Health Service Line
Atlanta VA Medical Center